simple_certmanager/admin.py

```diff
@@ -1,13 +1,89 @@
+from io import BytesIO
+from zipfile import ZipFile
+
 from django.contrib import admin
+from django.http import HttpResponse, HttpResponseRedirect
+from django.urls import reverse
 from django.utils.translation import gettext_lazy as _
 
 from privates.admin import PrivateMediaMixin
 
 from .forms import CertificateAdminForm
-from .models import Certificate
+from .models import Certificate, SigningRequest
 from .utils import suppress_cryptography_errors
 
 
+def download_csr(modeladmin, request, queryset):
+    if (num := len(queryset)) > 1:
+        zip_file = BytesIO()
+        with ZipFile(zip_file, "w") as zipf:
+            for csr in queryset:
+                csr_content = csr.csr
+                csr_filename = f"{csr.common_name}_csr.pem"
+                zipf.writestr(csr_filename, csr_content)
+        zip_file.seek(0)
+        response = HttpResponse(zip_file, content_type="application/zip")
+        response["Content-Disposition"] = 'attachment; filename="csr.zip"'
+        return response
+    elif num == 1:
+        csr = (first := queryset[0]).csr
+        response = HttpResponse(csr, content_type="application/x-pem-file")
```

Comment:
could probably also use the `FileResponse` approach if you wrap it in a `StringIO` object.

The `FileResponse` also infers the content type header from the filename, so might remove more boilerplate code:

```py
>>> import mimetypes
>>> mimetypes.guess_type("foo.pem")
('application/pem-certificate-chain', None)
```
seems to be fine on my system :)